### DISCUSSION TITLE
OpenNIC MAC address varies on each insmod of onic driver

### DISCUSSION THREAD

**Discussion by attdone** [Xilinx/open-nic-driver#50] [Q&A]
Hi Team, 
I am currently using OpenNIC in Alveo u200, as I could see the MAC address are changing everytime.
At first, **insmod onic.ko**
The MAC value of interfaces **enp8s0f0**, **enp8s0f1** were **00:0a:35:e9:e5:68** and **00:0a:35:87:5b:ee** respectively. 
I removed the driver using **rmmod onic** and restarted the driver after some seconds, insmod onic.ko. At this moment the MAC address is different i.e., **00:0a:35:fb:a7:0c** and **00:0a:35:6a:0c:32** for **enp8s0f0** and **enp8s0f1**.

When I run TestPMD DPDK command (after _dpdk-devbind.py -b vfio-pci enp8s0f0_) I could see the MAC address is **15:16:17:18:19:1A**. Why the MAC didnot get updated?

Please let me know why the MAC keeps changing and let me how to fix this issue. Any suggestions would be highly appreciated.
Thanks in advance.

**Comment by cneely-amd** [comment]
There's a section within onic_main.c, where it assigns a random value for the source MAC address (search for "random").  You can edit this to be some other value:
```
	memset(&saddr, 0, sizeof(struct sockaddr));
	memcpy(saddr.sa_data, onic_default_dev_addr, 6);
	get_random_bytes(saddr.sa_data + 3, 3);
	onic_set_mac_address(netdev, (void *)&saddr);
```
